src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+
+        Exception exception = assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user2);
+        });
+        assertEquals("Користувач з таким логіном вже існує", exception.getMessage());
+    }
+
+    @Test
+    void register_loginTooShort_notOk() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("password");
+        user.setAge(20);
+
+        Exception exception = assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals("Логін повинен містити щонайменше 6 символів", exception.getMessage());
+    }
+
+    @Test
+    void register_passwordTooShort_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("short");
+        user.setAge(20);
+
+        Exception exception = assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals("Пароль повинен містити щонайменше 6 символів", exception.getMessage());
+    }
+
+    @Test
+    void register_ageTooYoung_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(17);
+
+        Exception exception = assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals("Користувач повинен бути не молодшим за 18 років", exception.getMessage());
+    }
+
+    @Test
+    void register_validUser_ok() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser);
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+    }
+
+    @Test
+    void register_passwordEdgeCases_notOk() {
+        User user = new User();
+
+        user.setLogin("uniqueLogin1");
+        user.setPassword("");
+        user.setAge(20);
+        Exception exception1 = assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(user));
+        assertEquals("Пароль повинен містити щонайменше 6 символів", exception1.getMessage());
+
+        user.setLogin("uniqueLogin2");
+        user.setPassword("abc");
+        Exception exception2 = assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(user));
+        assertEquals("Пароль повинен містити щонайменше 6 символів", exception2.getMessage());
+
+        user.setLogin("uniqueLogin3");
+        user.setPassword("abcdf");
+        Exception exception3 = assertThrows(InvalidUserDataException.class, () ->
+                registrationService.register(user));
+        assertEquals("Пароль повинен містити щонайменше 6 символів", exception3.getMessage());
+    }
```

Comment:
Great job on testing the edge cases for password length. However, you're missing the test case for a password that is exactly 6 characters long. This is an important edge case to test.